src/librustc/middle/ty.rs

```diff
@@ -5459,6 +5465,11 @@ impl<'tcx> ctxt<'tcx> {
                         &format!("consider boxing your closure and/or \
                                   using it as a trait object"));
                 }
+            },
+            terr_ty_param_default_mismatch(expected) => {
+                self.sess.span_note(sp,
+                    &format!("found conflicting defaults {:?} {:?}",
```

Comment:
ok yeah we can do better than this, as you suggested on IRC :) 

I think the best thing would be if we recorded some sort of *origin* for every type variable -- ideally, this would be a span (in the current fn) and something like "type parameter X of fn Y", but maybe we just start with a span -- and then we could show the user where the defaults came from.